sympy/functions/special/error_functions.py

```diff
@@ -1694,7 +1694,7 @@ class Li(Function):
     def eval(cls, z):
         if z is S.Infinity:
             return S.Infinity
-        elif z == S(2):
+        elif z is Integer(2):
```

Comment:
Don't use `is` to compare things unless they are `S.something` (S(2) is not singletonized in SymPy). Also this can just be `z == 2`. The type of the rhs doesn't matter for `==`. `S(2) == 2`, `S(2) == 2`, and `2 == 2` all give True.

Reply:
Correct, but `z == 2`  will call sympify one way or the other, so providing that is better.

No really sure why it works, but even `Integer(1) + Integer(1) is S(2)` returns True. And it seems faster than ==.

Reply:
> Correct, but z == 2 will call sympify one way or the other, so providing that is better.

Not sure what you are saying here, but I would say `z == 2` is better because it is more readable. The `Integer` is unnecessary.

> No really sure why it works, but even Integer(1) + Integer(1) is S(2) returns True. And it seems faster than ==.

It may work due to the cache, but you shouldn't rely on it. is comparisons are only guaranteed to work for singletonized objects. And in those cases it would be good to be explicit by using `S.whatever`. 

On a related note, `2 is 2` will work in CPython but won't work in other implementations like PyPy. Basically, just because `is` happens to work in an example doesn't mean it will always work, and it can lead to weird and hard to reproduce bugs when it doesn't. It's unfortunate. I really wish Python didn't have the `is` operator, because all it does is trip people up who don't realize that it isn't actually equality testing.